lib/api.authentication.js

```diff
@@ -289,3 +289,29 @@ var shibbolethTenantCallback = module.exports.shibbolethTenantCallback = functio
 var ldapLogin = module.exports.ldapLogin = function(restCtx, username, password, callback) {
     RestUtil.RestRequest(restCtx, '/api/auth/ldap', 'POST', {'username': username, 'password': password}, callback);
 };
+
+/**
+ * Get a secret token to be used to reset password
+ *
+ * @param  {RestContext}    restCtx         Standard REST Context object that contains the current tenant URL
+ * @param  {String}         username        The username for which a password needs to be reset
+ * @param  {Function}       callback        Standard callback method
+ * @param  {Object}         callback.err    An error that occurred, if any
+ */
+var getResetPasswordSecret = module.exports.getResetPasswordSecret = function(restCtx, username, callback) {
+    RestUtil.RestRequest(restCtx, '/api/auth/local/reset/init/' + username, 'GET', null, callback);
```

Comment:
I'd suggest `/api/auth/local/reset/password-secret` since REST endpoints' resources should be nouns and not verbs.. also, we don't need the `/` do we?